src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -653,6 +663,33 @@ public void finish() {
             return coverageDistribution;
         }
 
+        /** Emits a file with per base coverage if an output file has been set. */
+        private void emitPerBaseCoverageIfRequested() {
+            if (this.perBaseOutput == null) return;
```

Comment:
I totally agree, my main goal here was to keep the `finish()` method from getting much longer or more convoluted, hence the `ifRequested()` in the name to indicate that it might not do anything.  I sorely miss scala when working in Java these days, and `java.util.Optional` isn't really well enough integrated/idiomatic enough to be a good replacement to scala's `Option`.